### PR TITLE
Ok http updates

### PR DESCRIPTION
**Pull request by sebaraba**

### CODE REVIEW

File: src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/EnclaveIpcService.kt [hunk 1]
```diff
@@ -2,62 +2,56 @@ package org.web3j.quorum.enclave.protocol.ipc
...
+import org.web3j.quorum.enclave.EnclaveClientConnectionException
 import org.web3j.quorum.enclave.protocol.EnclaveService
-import org.web3j.quorum.enclave.protocol.utils.RequestBuilder
-import org.web3j.quorum.enclave.protocol.utils.ResponseParser
 
 /**
  * IPC service layer
  */
-abstract class EnclaveIpcService : EnclaveService {
+class EnclaveIpcService(private val url: String, private val port: Int, private val client: OkHttpClient = OkHttpClient()) : EnclaveService {
```
Comment (nicolae-leonte-go) [inline src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/EnclaveIpcService.kt]: What is the purpose of the url and port for the IpcService?
As far as I can see it is the user's concern to initialize an OkHttpClient with a UnixDomainSocketFactory. Was that the intention? If it was - I suspect the EnclaveHttpService could also be used with a user initialized OkHttpClient.

File: src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/EnclaveIpcService.kt [hunk 2]
```diff
@@ -2,62 +2,56 @@ package org.web3j.quorum.enclave.protocol.ipc
...
-     * Perform IO read/write
-     */
-    private fun performIO(data: String): String {
-        val io = getIO()
-        io.write(data)
-
-        val response = io.read()
-
-        return ResponseParser.parseChunkedResponse(response)
+        val client = OkHttpClient()
```
Comment (nicolae-leonte-go) [inline src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/EnclaveIpcService.kt]: The client passed to the constructor be used here.

File: src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/UnixDomainSocketFactory.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import java.net.InetSocketAddress
+import java.net.Socket
+import java.net.SocketAddress
+import javax.net.SocketFactory
+import jnr.unixsocket.UnixSocket
+import jnr.unixsocket.UnixSocketAddress
+import jnr.unixsocket.UnixSocketChannel
+
+/** Impersonate TCP-style SocketFactory over UNIX domain sockets. */
+class UnixDomainSocketFactory: SocketFactory() {
```
[the remark this answers is not in the document]
Comment (sebaraba) [reply src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/UnixDomainSocketFactory.kt]: I just realized we had the same functionality inside web3j already. I ditched this kotlin implementation.

File: src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/EnclaveIpcService.kt (hunk 1 above)
Comment (sebaraba) [reply src/main/kotlin/org/web3j/quorum/enclave/protocol/ipc/EnclaveIpcService.kt]: As far as I know okHttp only work using an uri. So when we want to send a new request to Constellation we need `url` and `port`.

**Pull request closed without merging**